gen/encoder.go

```diff
@@ -195,7 +214,7 @@ func (g *Generator) notEmptyCheck(t reflect.Type, v string) string {
 	}
 
 	switch t.Kind() {
-	case reflect.Slice, reflect.Map:
+	case reflect.Slice, reflect.Map: // note: Array types don't have a useful empty value
```

Comment:
I think it's better to keep the comment for Array closer to the actual code path it takes.